prow/config/config.go

```diff
@@ -38,6 +38,15 @@ type Config struct {
 
 	Plank    Plank     `json:"plank,omitempty"`
 	Triggers []Trigger `json:"triggers,omitempty"`
+
+	// ProwJobNamespace is the namespace in the cluster that prow
+	// components will use for looking up ProwJobs. The namespace
+	// needs to exist and will not be created by prow.
+	ProwJobNamespace string `json:"prowjob_namespace,omitempty"`
```

Comment:
@spxtr I am open to suggestions here as to how to lay these out. As we move forward and evolve the config, it seems to me that we will want to group the config into {k8s, github, prow}-specific options.